### DISCUSSION TITLE
Developing Kong custom TCP plugin

### DISCUSSION THREAD

**Discussion by rohank2002** [Kong/kong#13911] [Help]
Hello, we have a use case where we will have to develop a custom TCP kong plugin which can terminate requests if the client certificate presented during the TLS handshake with the server(kong is just a passthrough) does not match a certain pattern. For this we will have to decode the TCP data and validate based on its details. Based on the comment [here](https://github.com/Kong/kong/issues/6679#issuecomment-1359896560), seems like we would not be able get access to the TCP stream at the plugin level. Can someone please confirm this ? If it is possible to access the TCP data stream at the plugin level, then are there any other TCP kong plugin available which decodes TCP packet (for our reference)?

**Comment by Water-Melon** [comment]
Is your requirement to match the certificate? If so, could you consider implementing this functionality in the plugin's certificate phase handler?
